plugins/jokes.py

```diff
@@ -1,95 +1,99 @@
-import codecs
-import os
 import random
+from pathlib import Path
 
 from cloudbot import hook
 
+joke_lines = {}
+
+
+def load_joke_file(path):
+    """Loads all the lines from a file, excluding blanks and lines that have been 'commented out'.
+    :type path: Path
+    :rtype: List[str]
+    """
+    with path.open(encoding='utf-8') as f:
+        return [line.strip() for line in f
+                if line.strip()
+                and not line.startswith('//')]
+
 
 @hook.on_start()
 def load_jokes(bot):
-    """
+    """Load strings into memory from files in the data directory.
+    Put 'NAME.txt' in `file_list` to make those strings available as `joke_lines['NAME']`.
     :type bot: cloudbot.bot.Cloudbot
     """
-    global yo_momma, do_it, pun, confucious, one_liner, wisdom, book_puns, lawyerjoke, kero_sayings
-
-    with codecs.open(os.path.join(bot.data_dir, "yo_momma.txt"), encoding="utf-8") as f:
-        yo_momma = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "do_it.txt"), encoding="utf-8") as f:
-        do_it = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "puns.txt"), encoding="utf-8") as f:
-        pun = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "confucious.txt"), encoding="utf-8") as f:
-        confucious = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "one_liners.txt"), encoding="utf-8") as f:
-        one_liner = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "wisdom.txt"), encoding="utf-8") as f:
-        wisdom = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "book_puns.txt"), encoding="utf-8") as f:
-        book_puns = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "lawyerjoke.txt"), encoding="utf-8") as f:
-        lawyerjoke = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "kero.txt"), encoding="utf-8") as f:
-        kero_sayings = [line.strip() for line in f.readlines() if not line.startswith("//")]
+    global joke_lines
+    data_directory = Path(bot.data_dir)
+    file_list = [
+        'yo_momma.txt',
+        'do_it.txt',
+        'puns.txt',
+        'confucious.txt',
+        'one_liners.txt',
+        'wisdom.txt',
+        'book_puns.txt',
+        'lawyerjoke.txt',
+        'kero.txt',
+    ]
+    for file_name in file_list:
+        file_path = data_directory / file_name
+        joke_lines[file_path.stem] = load_joke_file(file_path)
```

Comment:
Perhaps a dynamic system similar to `foods.py` would work well here, as each command behaves pretty much the same. Up to you though, there isn't as much to deal with as there was in `foods.py` so it's not as much of a gain.

Reply:
Maybe another time. I've identified a handful of plugins (eightball, fmk, fortune, kenm, topicchange, verysmart) that all have a similar behaviour and combined together it might be worthwhile.

Reply:
yeah, I've had those on my todo list to investigate merging them in to a larger dynamic plugin but haven't gotten around to it.

Reply:
I'll assemble a list and create an issue to discuss it.